zio-kafka/src/main/scala/zio/kafka/admin/AdminClient.scala

```diff
@@ -627,33 +628,43 @@ object AdminClient {
      */
     override def listConsumerGroupOffsets(
       groupSpecs: Map[String, ListConsumerGroupOffsetsSpec]
-    ): Task[Map[TopicPartition, OffsetAndMetadata]] =
+    ): Task[Map[String, Map[TopicPartition, OffsetAndMetadata]]] =
       fromKafkaFuture {
         ZIO.attemptBlocking(
           adminClient
             .listConsumerGroupOffsets(groupSpecs.map { case (groupId, offsetsSpec) =>
               (groupId, offsetsSpec.asJava)
             }.asJava)
-            .partitionsToOffsetAndMetadata()
+            .all()
         )
       }
-        .map(_.asScala.filter { case (_, om) => om ne null }.toMap.bimap(TopicPartition(_), OffsetAndMetadata(_)))
+        .map(
+          _.asScala.map { case (groupId, offsets) =>
+            groupId -> offsets.asScala.filter { case (_, om) => om ne null }.toMap
+              .bimap(TopicPartition(_), OffsetAndMetadata(_))
+          }.toMap
+        )
 
     override def listConsumerGroupOffsets(
       groupSpecs: Map[String, ListConsumerGroupOffsetsSpec],
       options: ListConsumerGroupOffsetsOptions
-    ): Task[Map[TopicPartition, OffsetAndMetadata]] =
+    ): Task[Map[String, Map[TopicPartition, OffsetAndMetadata]]] =
       fromKafkaFuture {
         ZIO.attemptBlocking(
           adminClient
             .listConsumerGroupOffsets(
               groupSpecs.map { case (groupId, offsetsSpec) => (groupId, offsetsSpec.asJava) }.asJava,
               options.asJava
             )
-            .partitionsToOffsetAndMetadata()
+            .all()
         )
       }
-        .map(_.asScala.filter { case (_, om) => om ne null }.toMap.bimap(TopicPartition(_), OffsetAndMetadata(_)))
+        .map(
+          _.asScala.map { case (groupId, offsets) =>
+            groupId -> offsets.asScala.filter { case (_, om) => om ne null }.toMap
+              .bimap(TopicPartition(_), OffsetAndMetadata(_))
```

Comment:
If efficiency is very important here, we should do the transformation (done by bimap) before we convert the collection into a map.

In that case it could be as simple as:
```suggestion
            groupId -> offsets.asScala
              .filter { case (_, om) => om ne null }
              .bimap(TopicPartition(_), OffsetAndMetadata(_))
              .toMap
```
(And we need to define a new bimap that works on `List[(A,B)]`.)

Reply:
Done